catalog/resource_share.go

```diff
@@ -128,19 +120,23 @@ func (si ShareInfo) resourceShareMap() map[string]SharedDataObject {
 	return m
 }
 
-func Equal(a, b SharedDataObject) bool {
-	if b.SharedAs == "" {
-		b.SharedAs = a.SharedAs
+func (sdo SharedDataObject) Equal(other SharedDataObject) bool {
+	if other.SharedAs == "" {
+		other.SharedAs = sdo.SharedAs
 	}
-	return reflect.DeepEqual(a, b)
+	//don't compare computed fields
+	other.AddedAt = sdo.AddedAt
```

Comment:
this will pollute the state of `other`, leading to unexpected behavior

Reply:
I thought `other` is passed by value, so won't be modified outside the function?